pkg/storage/export/export/export_test.go

```diff
@@ -844,7 +844,7 @@ var _ = Describe("Export controller", func() {
 			Expect(ok).To(BeTrue())
 			pod, ok := create.GetObject().(*k8sv1.Pod)
 			Expect(ok).To(BeTrue())
-			Expect(pod.GetName()).To(Equal(fmt.Sprintf("%s-%s", exportPrefix, testVMExport.Name)))
+			Expect(pod.GetName()).To(Equal(controller.getExportPodName(testVMExport)))
```

Comment:
this should probably be `controller.getExportLabelValue(testVMExport)`
Maybe also ensure that the length of pod.GetName() <= validation.DNS1035LabelMaxLength

Reply:
As in, the test only? Or where the pod name is set in the template as well? The latter is in a different package IIRC, so I’d have to duplicate some logic or move the method call to a functions in a library available to both packages.

Reply:
Ah, I see the test logs now. I’ll take a closer look in the AM

Reply:
@awels

The problem here seems to be that getExportPodName uses DNS1035LabelMaxLength (i.e. <= 63)
https://github.com/kubevirt/kubevirt/blob/main/pkg/storage/export/export/export.go#L770

But when Rendering the pod manifest, we use `naming.GetPodName` which uses DNS1123SubdomainMaxLength (i.e. <= 253). https://github.com/openshift/library-go/blob/master/pkg/build/naming/namer.go#L41

The rendered value ends up being <= 253, so it's passed through as is. 

The naming package seems to align with k8s upstream https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L254

This could go either way, but for now I'll force the renderer to use the shorter length. Let me know if you want me to go the other way (and change the getExportPodName to use DNS1123SubdomainMaxLength)

Reply:
https://github.com/kubevirt/kubevirt/commit/114474c1020d526625afc1ddea1a955331f6004c should contain the fix as described, as well as the length assertion in testing.